### PR TITLE
Added support for react style mouse events

### PR DESCRIPTION
**Pull request by morten-olsen**
Since the only way I could find for using mouse events was to use componentDidMount and refs

```javascript
class SomeComp extends Component {
  componentDidMount() {
    this.refs.box.on('mousedown', () => {
      console.log('stuff');
    })
  }
  render() {
    return (
      <box
        ref="box"
      />
    )
  }
}
```

I have therefore extended `ReactBlessedComponent` to support react style mouse events like

```javascript
export default () => (
  <box onMouseDown={() => console.log('stuff')} />
)
```

The complete list of supported events are:

* mousedown
* mouseup
* mousemove
* mouse
* click

The list can be expanded in the `mouseEvents` object in `ReactBlessedComponent` simply by adding a new property where the name is the `on` event name and the value is the `prop` name for the component

### CODE REVIEW

**Comment by Yomguithereal** [comment]
I am a bit unclear about the reasons why the mouse events don't work in the first place. I'll need to investigate a bit.

File: src/ReactBlessedComponent.js [hunk 1]
```diff
@@ -52,6 +61,11 @@ export default class ReactBlessedComponent {
         handler(...args);
       }
     };
+    this._mouseEvents = mouseEventNames.reduce((output, eventName) => {
```
Comment (Yomguithereal) [inline src/ReactBlessedComponent.js]: Please use a basic loop here for performance reasons.

File: src/ReactBlessedComponent.js [hunk 2]
```diff
@@ -168,6 +185,9 @@ export default class ReactBlessedComponent {
     const node = ReactBlessedIDOperations.get(this._rootNodeID);
 
     node.off('event', this._eventListener);
+    mouseEventNames.forEach(eventName => {
+      node.on(eventName, this._mouseEvents[eventName]);
```
Comment (Yomguithereal) [inline src/ReactBlessedComponent.js]: Shouldn't this be a `.off` rather than a `.on`?

File: src/ReactBlessedComponent.js [hunk 3]
```diff
@@ -185,6 +205,12 @@ export default class ReactBlessedComponent {
   getPublicInstance() {
     return ReactBlessedIDOperations.get(this._rootNodeID);
   }
+
+  _handleMouseEvent(propName, ...args) {
```
Comment (Yomguithereal) [inline src/ReactBlessedComponent.js]: Can you remind me why we need to handle mouse events separately? Doesn't bless emit the mouse event through the generic handler? Maybe we need to open an issue with blessed?

**Comment by morten-olsen** [comment]
As far as i could see, blessed handles mouse event differently than all other events, and does not emit them using the default `.on('event', `, as it does with other events, but only using the actual named events

**Comment by morten-olsen** [comment]
I'll implement the change requested, later today

**Comment by morten-olsen** [comment]
But yes, it might be a blessed issue. I'll try taking a closer look at their event handling and see if there is a reason for those not being emittet through the generic handler

**Pull request closed without merging**